### PR TITLE
Fix ITCrossEngineDremio

### PR DESCRIPTION
**Pull request by XN137**
- add query helpers
- improve error handling with additional info about SQL
- Dremio context param no longer works, so we need to use fully
  qualified table names
- Flink needs to create database if it doesnt exist

### CODE REVIEW

File: nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java [hunk 1]
```diff
@@ -38,29 +39,41 @@
...
   }
 
+  public String getCatalogName() {
+    return catalogName;
+  }
+
   private String createPayload(String query) throws JsonProcessingException {
     Map<String, Object> payload = new HashMap<>();
     payload.put("sql", query);
+    // TODO: for some reason this no longer works and we have to use fully qualified table names
```
Comment (snazy) [inline nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java:76]: What's the "TODO" here?

File: nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java [hunk 2]
```diff
@@ -102,17 +115,16 @@ private String performHttpRequest(String url, String payload) throws IOException
   private void waitForJobCompletion(String jobId, String query) throws IOException {
     // The doc for getting the job status for cloud is not there, but it is similar to software
     // See docs: https://docs.dremio.com/software/rest-api/jobs/get-job/
-    String jobState = "RUNNING";
-    String url = baseUrl + "/v0/projects/" + projectId + "/job/" + jobId;
+    String url = projectUrl + "/job/" + jobId;
```
Comment (snazy) [inline nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java:118]: Looks like it's worth to refactor this concatenation from three places into a helper method

File: nqeit-cross-engine/src/test/java/org/projectnessie/integtests/crossengine/ITCrossEngineDremio.java [hunk 3]
```diff
@@ -68,23 +89,38 @@ public static void setupEngines(
...
+        .forEach(x -> sparkSql("DROP TABLE IF EXISTS %s", sparkTable(x)));
+  }
+
+  private static String sparkTable(String tableName) {
+    String sparkCatalogName = "nessie"; // see IcebergSparkExtension
+    return String.join(".", sparkCatalogName, NAMESPACE, tableName);
+  }
+
+  private static String dremioTable(String tableName) {
+    return Stream.of(dremioHelper.getCatalogName(), NAMESPACE, tableName)
```
Comment (snazy) [inline nqeit-cross-engine/src/test/java/org/projectnessie/integtests/crossengine/ITCrossEngineDremio.java:102]: String.format() is probably easier to read

File: nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java [hunk 4]
```diff
@@ -38,29 +39,41 @@
...
+          "Failed to parse Response from " + url + " as JSON:\n" + json, e);
+    }
   }
 
-  DremioHelper(String projectId, String token, String baseUrl, String catalogName) {
-    this.projectId = projectId;
+  DremioHelper(String token, String baseUrl, String projectId, String catalogName) {
     this.token = token;
-    this.baseUrl = baseUrl;
+    if (baseUrl.contains("://app.")) {
```
[the remark this answers is not in the document]
Comment (XN137) [reply nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java]: its for user convenience, moved to IcebergDremioExtension with additional comments

File: nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java (hunk 2 above)
Comment (XN137) [reply nqeit-iceberg-dremio-extension/src/main/java/org/projectnessie/integtests/dremio/DremioHelper.java]: only used in 2 places, but ok

**Review by snazy — APPROVED** [review]

**Pull request merged**